src/mugs.js

```diff
@@ -170,15 +170,33 @@ define([
                 }
             }
 
-            return this.messages.update(attr, {
+            var changed = false;
+
+            if (attr === "nodeID" && mug.__originalNodeID) {
+                var info = "";
+                if (!message && mug.p.nodeID.toLowerCase() !== mug.__originalNodeID) {
+                    info = gettext(
+                        "Changing a Question ID will create a new Question ID (and a new data column). " +
+                        "It will NOT update the existing or previously submitted data.");
+                }
+                changed = this.messages.update(attr, {
+                    key: "mug-nodeID-info",
+                    level: this.INFO,
+                    message: util.format(info || "", {question: label})
+                }) || changed;
```

Comment:
This should be moved into `nodeID.validateFunc`

https://github.com/dimagi/Vellum/blob/3f83f74a4b5fc0bb57c236c15e057a3ce13f1302/src/mugs.js#L846-L871

Also, a test should be added to ensure that the message is present when the `nodeID` is changed. I would probably add that test to _tests/form.js_ since most mug validation tests are in there. Here's an example you can follow (but pass something like `HIDDEN_VALUE_IN_GROUP_XML` to `util.loadXML()`—or if that form doesn't work well, you can add a new one for this test):

https://github.com/dimagi/Vellum/blob/3f83f74a4b5fc0bb57c236c15e057a3ce13f1302/tests/form.js#L170-L177